pkg/process/checks/net.go

```diff
@@ -126,12 +133,81 @@ func (c *ConnectionsCheck) Realtime() bool { return false }
 // ShouldSaveLastRun indicates if the output from the last run should be saved for use in flares
 func (c *ConnectionsCheck) ShouldSaveLastRun() bool { return false }
 
-// Run runs the ConnectionsCheck to collect the active network connections
-// and any closed network connections since the last Run.
-// For each connection we'll return a `model.Connection`
-// that will be bundled up into a `CollectorConnections`.
-// See agent.proto for the schema of the message and models.
-func (c *ConnectionsCheck) Run(nextGroupID func() int32, _ *RunOptions) (RunResult, error) {
+func (c *ConnectionsCheck) runGRPC(nextGroupID int32) (RunResult, error) {
+	// Create a context with a timeout of 10 seconds
+	timedContext, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(func(_ context.Context, _ string) (nnet.Conn, error) {
+			return nnet.Dial(net.NetType, c.syscfg.GRPCSocketFilePath)
+		}),
+		grpc.WithInsecure(),
+	}
+
+	// the second parameter is unused because we use a custom  dialer that has the state to see the address.
+	conn, err := grpc.DialContext(timedContext, "unused", opts...)
+	if err != nil {
+		return nil, fmt.Errorf("unable to dial with context to grpc server due to: %v", err)
+	}
+	defer conn.Close()
+	client := connectionserver.NewSystemProbeClient(conn)
+
+	response, err := client.GetConnections(timedContext, &connectionserver.GetConnectionsRequest{ClientID: c.tracerClientID}, grpc.MaxCallRecvMsgSize(api.MaxGRPCServerMessage))
+	if err != nil {
+		return nil, fmt.Errorf("unable to get connections from grpc server due to: %v", err)
+	}
+
+	// We only need to fetch once for a batch, and this boolean indicates that.
+	processFetchSucceeded := true
+	if err := c.processData.Fetch(); err != nil {
+		processFetchSucceeded = false
+		log.Warnf("error collecting processes for filter and extraction: %s", err)
+	}
+
+	unmarshaler := netEncoding.GetUnmarshaler(netEncoding.ContentTypeProtobuf)
+	var batchMessages []model.MessageBody
+
+	for {
+		start := time.Now()
+		res, err := response.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("unable to get respinse from grpc server due to: %v", err)
+		}
+
+		batch, err := unmarshaler.Unmarshal(res.Data)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal response due to: %v", err)
+		}
+		log.Debugf("[grpc] the number of connections in grpc is %d", len(batch.Conns))
+		if !processFetchSucceeded {
+			c.dockerFilter.Filter(batch)
+		}
+
+		// Resolve the Raddr side of connections for local containers
+		LocalResolver.Resolve(batch)
```

Comment:
This may not resolve everything. We use the connections to "resolve" container ids for each other, so there is a chance some connections will be missing in this batch that would have previously been there and used for resolution. I think you are going to have to collect all batches and then do resolution on them once as was done before.

Reply:
Maybe we can limit for now the process-agent changes to just accumulating batches for now and then running the existing code path over all the connections? I think these are tricky changes that may lead to us missing something. We should still see some memory improvements on the system-probe side because of the batching. process-agent changes can be done in a follow up.

Reply:
> Maybe we can limit for now the process-agent changes to just accumulating batches for now and then running the existing code path over all the connections

I second this suggestion.